### PR TITLE
[docs] platform+azure configuration updates

### LINKED ISSUE #7955: [Docs] Add instructions for Azure - Shared Image Gallery + Private DNS Zone
https://docs.google.com/document/d/1fMgpAg9YoatzuQXfHjPOqZyP_CkYkLgi92M-FQfC4_g/edit#heading=h.ojykwrjh2kfr

### PR DESCRIPTION
**Pull request by polarweasel**
Fixes #7955

**Preview build** is here: https://deploy-preview-8005--infallible-bardeen-164bc9.netlify.app/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure/

### CODE REVIEW

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md [hunk 1]
```diff
@@ -65,53 +65,118 @@ showAsideToc: true
...
+
+## Configure Azure
 
 ![Prepare Azure cloud to install Yugabyte Platform](/images/yb-platform/install/azure/platform-azure-prepare-cloud-env-4.png)
 
-#### Provider name
-This is an internal tag used for organizing your providers, so you know where you want to deploy your YugabyteDB universes.
+**Provider name** is an internal tag used for organizing your providers, so you know where you want to deploy your YugabyteDB universes.
+
+**Subscription ID**. You need an Azure subscription ID for cost management. The virtual machine resources managed by the Platform will be tagged with this subscription.
```
Comment (WesleyW) [inline docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md:89]: Change format to match other ones - "Subscription ID is...."

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md [hunk 2]
```diff
@@ -65,53 +65,118 @@ showAsideToc: true
...
+
+**Resource group** is the group in which YugabyteDB nodes compute and network resources will be created. Your Azure active directory application (clientId and client secret) needs to have `Network Contributor` and `Virtual Machine Contributor` roles assigned for this resource group.
```
Comment (WesleyW) [inline docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md:91]: Capitalize "Active Directory"

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md [hunk 3]
```diff
@@ -65,53 +65,118 @@ showAsideToc: true
...
+
+**Tenant ID** is the Azure Active Directory tenant ID which belongs to an active subscription. To find your tenant ID, follow the instructions on [this Microsoft Azure page](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#get-tenant-and-app-id-values-for-signing-in).
+
+**Client ID** is the client ID of an application registered in your Azure Active Directory.
```
Comment (WesleyW) [inline docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md:95]: I know it's the same destination as above, but there's another section to generate/find the client ID/secret that could also be linked here. https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal#option-2-create-a-new-application-secret

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md [hunk 4]
```diff
@@ -65,53 +65,118 @@ showAsideToc: true
...
+
+**Client secret** is the client secret of an application registered in your Azure Active Directory.
+
+**Private DNS zone** lets you use a custom domain name for the nodes in your universe. Refer to [Set a private DNS zone](#set-a-private-dns-zone-optional) for instructions on how to set this up.
+
+**Virtual network setup**. Using your custom Virtual Network is supported. This allows you the highest level of customization for your network setup.
```
Comment (WesleyW) [inline docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md:101]: So I don't know if this info is relevant or not, but currently, the user has to provide their own virtual network. We haven't enabled the auto-create option yet.

**Review by WesleyW — APPROVED** [review]
LGTM! Couple comments on some minor changes that could be made. The images still look big on my end, but like you said, it can't be fixed here.

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md (hunk 4 above)
Comment (polarweasel) [reply docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/azure.md:101]: I think I'll leave this one as-is...

**Comment by polarweasel** [comment]
Wesley approved the fixes in slack... time to merge :)

**Pull request merged**